src/highdicom/spatial.py

```diff
@@ -35,6 +35,395 @@ def create_rotation_matrix(
     ])
 
 
+def _create_affine_transformation_matrix(
+    image_position: Sequence[float],
+    image_orientation: Sequence[float],
+    pixel_spacing: Sequence[float],
+) -> np.ndarray:
+    """Create affine matrix for transformation.
```

Comment:
In case we ever have to revisit this again (you may have to fix my spacing here as the github interface makes that fiddly):

```suggestion
    """Create affine matrix for transformation.
    
    The resulting transformation matrix maps the center of a pixel identified by zero-based 
    integer indices into the frame of reference, i.e. an input value of (0, 0) represents the
    center of the top left hand corner pixel.
    
    See :dcm:`Equation C.7.6.2.1-1 <part03/sect_C.7.6.2.html#sect_C.7.6.2.1.1>`.
    
```